File: 파이썬/2587.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/2587
'''

# 버블 정렬
l = []
for i in range(5) :
    m = int(input())
    l.append(m)

for i in range(4, -1, -1) :
    for j in range(0, i) :
        if l[j] > l[j+1] :
            tmp = l[j]
            l[j] = l[j+1]
            l[j+1] = tmp

print(sum(l) // 5)
print(l[2])
    
             
# 선택 정렬
l = []

for i in range(5) :
    m = int(input())
    l.append(m)

for i in range(4) :
    key = i
    for j in range(i+1, 5) :
         if l[key] > l[j] :  
             key = j 
    tmp = l[key]
    l[key] = l[i]
    l[i] = tmp
         
print(sum(l) // 5)
print(l[2])
    
# 삽입 정렬
l = []

for i in range(5) :
    m = int(input())
    l.append(m)

for i in range(1, 5) :
    key = l[i] 
    j = i-1 
    while j>=0 and l[j] > key :
        l[j+1] = l[j]
        j -= 1 
    l[j+1] = key
    
print(sum(l) // 5)
print(l[2]) 

File: C++/1.1 정렬 - 초급/2587.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/2587
*/

// 버블 정렬
#include <iostream>

using namespace std;

int arr[5];
int sum;

int main(){

    for(int i=0; i<5; i++){
        cin >> arr[i];
        sum += arr[i];
    }    

    for(int i=4; i>=0; i--){
        for(int j=0; j<i; j++){
            if(arr[j] > arr[j+1]){
                int tmp = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = tmp;
            }
        }
    }

    cout << sum / 5 << "\n";
    cout << arr[2];
    
    return 0;
}

// 선택 정렬
#include <iostream>

using namespace std;

int arr[5];
int sum;

int main(){

    for(int i=0; i<5; i++){
        cin >> arr[i];
        sum += arr[i];
    }   
    
    for(int i=0; i<4; i++){
        int key = i;
        for(int j=i+1; j<5; j++){
            if(arr[key] > arr[j]){
                key = j;
            }
        }
        int tmp = arr[key];
        arr[key] = arr[i];
        arr[i] = tmp;
    }

    cout << sum / 5 << "\n";
    cout << arr[2];
    
    return 0;
}

// 삽입 정렬
#include <iostream>

using namespace std;

int arr[5];
int sum, j;

int main(){

    for(int i=0; i<5; i++){
        cin >> arr[i];
        sum += arr[i];
    }   
    
    for(int i=1; i<5; i++){ 
        int key = arr[i]; 
        for(j=i-1; j>=0 && arr[j] > key; j--){
            arr[j+1] = arr[j]; 
        } 
        arr[j+1] = key; 
    }

    cout << sum / 5 << "\n";
    cout << arr[2];
    
    return 0;
}